bughouse_console/src/private_persistence.rs

```diff
@@ -0,0 +1,57 @@
+use tide::utils::async_trait;
+use time::OffsetDateTime;
+
+#[derive(Debug, Clone, Copy)]
+pub enum RegistrationMethod {
+    Password,
+    GoogleOAuth,
+}
+
+impl RegistrationMethod {
+    pub fn to_string(self) -> String {
+        match self {
+            Self::Password => "Password",
+            Self::GoogleOAuth => "GoogleOAuth",
+        }.to_owned()
+    }
+    pub fn try_from_string(s: String) -> Option<Self> {
```

Comment:
Suggestion: return `Result` instead. This is in line with [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) and you wouldn't need to copy-paste `ok_or(anyhow::Error::msg("failed to parse registration method"))`.
Also I would include `s` into the error message.

Reply:
Done.